### PR TITLE
fix: add react-hooks eslint and fix missing deps/issues

### PR DESCRIPTION
**Pull request by chanceaclark**
## What
Add react-hooks eslint plugin and fix missing dependencies. Also fixed some logic around custom hooks.

### CODE REVIEW

**Comment by chanceaclark** [comment]
> a couple of missing newlines on some of the new files but other than that lgtm

Locally they have newlines 😖

**Comment by rtalvarez** [comment]
> > a couple of missing newlines on some of the new files but other than that lgtm
> 
> Locally they have newlines 😖

:fake-news:

File: packages/big-design/package.json [hunk 1]
```diff
@@ -22,7 +22,8 @@
     "build:es": "NODE_ENV=production BABEL_ENV=es babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/es",
     "build:dt": "tsc --emitDeclarationOnly",
     "ci": "yarn run lint && yarn run test --maxWorkers=2 && yarn run build",
-    "lint": "yarn run lint:tslint && yarn run lint:tsc && yarn run lint:prettier",
+    "lint": "yarn run lint:eslint && yarn run lint:tslint && yarn run lint:tsc && yarn run lint:prettier",
+    "lint:eslint": "eslint src/**/* --ext '.ts,.tsx' --fix",
```
Comment (deini) [inline packages/big-design/package.json]: Let's move this to husky 🐶

File: packages/big-design/src/hooks/useEventCallback.ts [hunk 2]
```diff
@@ -1,14 +1,15 @@
-import { useCallback, useEffect, useRef } from 'react';
+import { useCallback, useEffect, useRef, DependencyList } from 'react';
 
 type Callback<T> = (...args: T[]) => void;
 
-export function useEventCallback<T>(fn: Callback<T>, dependencies: any[]) {
+export function useEventCallback<T>(fn: Callback<T>, dependencies: DependencyList = []) {
```
Comment (deini) [inline packages/big-design/src/hooks/useEventCallback.ts]: Let's get rid of `dependencies` here.

**Pull request merged**